### PR TITLE
ci: build and deploy storybook

### PR DESCRIPTION
**Pull request by tonyo**
This PR adds a Travis job that builds Storybook component library (https://storybook.js.org/) for every branch and deploys the assets to GCS. Then, the storybooks can be accessible via https://storybook.getsentry.net/, one directory per branch. 
For example: https://storybook.getsentry.net/_master/index.html is for master branch.

A few caveats:
* Git branch name is slightly transformed: it's made lowercase, and all non-alphanumeric characters are either replaced (`.` and `/` become `-`) or removed. 
* There's no branch cleanup at the moment, but it can be added later: we additionally upload `build-info.json` file with some metadata (creation time, commit hash), or we could use GCS's lifecycle management. 

Initial feature request: https://github.com/getsentry/ops-req/issues/35

### CODE REVIEW

**Comment by JoshuaRLi** [comment]
There's going to be a bit of a merge conflict if this or #8419 is merged before the other. I'm currently refactoring make+travis, so I'll keep an eye on this PR.

**Comment by tonyo** [comment]
@JoshuaRLi thanks for the heads-up, I'll also keep your PR in mind.

File: .travis/deploy-storybook.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+GS_BUCKET_NAME=sentryio-storybook
+DEPLOY_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}
+echo "Build branch: ${DEPLOY_BRANCH}"
+
+# Transform the branch name to a bucket directory name, give preference to "master"
+if [ "${DEPLOY_BRANCH}" = "master" ]; then
+  BUCKET_DIR_NAME="_master"
+else
+  BUCKET_DIR_NAME=$(echo "${DEPLOY_BRANCH}" | tr '[:upper:]./' '[:lower:]--' | tr -cd '[:alnum:]-_')
```
Comment (billyvg) [inline .travis/deploy-storybook.sh]: Can you tell me what this does with the branch name? The common non-alphanumeric characters we use are `/`, `-`, `_`.

File: .travis/deploy-storybook.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -eu
+
+GS_BUCKET_NAME=sentryio-storybook
+DEPLOY_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}
+echo "Build branch: ${DEPLOY_BRANCH}"
+
+# Transform the branch name to a bucket directory name, give preference to "master"
+if [ "${DEPLOY_BRANCH}" = "master" ]; then
+  BUCKET_DIR_NAME="_master"
```
Comment (billyvg) [inline .travis/deploy-storybook.sh]: Why can't we have this be `master`?

**Review by billyvg — APPROVED** [review]
This looks great!

A few configuration things that are probably outside of this PR:

* Can we make `index.html` the directory index? e.g. `https://storybook.getsentry.net/_master/` should load `index.html`
* Can we also make `https://storybook.getsentry.net/` always load `master`'s version of storybook. We'll probably also want a way to list all the branches as well (maybe just move everything into another subdir?.)

File: .travis/deploy-storybook.sh (hunk 1 above)
Comment (billyvg) [reply .travis/deploy-storybook.sh]: Oops just re-read your PR

**Comment by tonyo** [comment]
@billyvg Thanks for checking, I simplified the bucket structure a bit, so now all branch uploads go to `/branches/BRANCH_NAME` directories. https://storybook.getsentry.net/branches/ now shows a list of all branches.
Regarding your questions about `index.html` and default branch: it will be added tomorrow 👍

**Pull request merged**